woocommerce/woocommerce-ios
# [Orders with Coupons M7]  Add informational tooltips

Closes: #10839. 
#10837 should be merged first


## Description
This PR adds informational tooltips on tapping the `(?)` symbol when adding coupons to an order, or when adding discounts to a product. This PR deals with the UI only, in the following we'll deal with the logic that makes coupons and discounts mutually exclusive.

| Orders | Orders tooltip |
|--------|--------|
| ![_coupons_tooltip](https://github.com/woocommerce/woocommerce-ios/assets/3812076/e9e86077-dbbe-4518-8549-c601774a6a4d) | ![Simulator Screenshot - iPhone 15 - 2023-10-10 at 20 19 42](https://github.com/woocommerce/woocommerce-ios/assets/3812076/1ab53dbe-8205-4d9b-841f-054fa8288374) | 

| Product | Product tooltip |
|--------|--------|
| ![_discounts_tooltip](https://github.com/woocommerce/woocommerce-ios/assets/3812076/e8ab8d3a-17ab-4655-94ff-4cb5660b01d3)  | ![Simulator Screenshot - iPhone 15 - 2023-10-10 at 20 20 02](https://github.com/woocommerce/woocommerce-ios/assets/3812076/9da74d4f-850e-4c7a-8c80-e54746e95542) | 


## Changes:
* Extracts the product discount row into a `@ViewBuilder` variable for cleaning up the main view. We also move all other `@ViewBuilder` vars to a private extension. 
* Adds `shouldShowCouponsInfoTooltip` state which will change when the `(?)` icon is tapped. This will show/hide an overlay with the informational tooltips.

## Testing instructions
* Switch `ordersWithCouponsM6` feature flag to true
* Go to `Orders` > `+` > Observe that:
  * `+ Add Coupon` is disabled. `(?)` is tappable. Tapping on it toggles the informational tooltip. Tapping again hides it.
  * Tap on `+ Add Product` and add a product to the order. Now `+ Add Coupon` is enabled, and the `(?)` icon is hidden.
  * On the product you've added to the order, or a new product added to the order, tap the chevron to expand the collapsable product row.
  * Observe that the `(?)` in `+ Add discount` is tappable, tapping it shows an informational tooltip and tapping again hides it.
  * Adding a discount to the product will hide both the `+ Add discount` and the informational tooltip icon.
